### ISSUE TITLE
Test time augmentations

### ISSUE THREAD

**Issue by anuj-sharma-19** [hht1996ok/EA-LSS#8]
Hi @hht1996ok,

Thanks for the great work, and releasing the code!

I have a query regarding the TTA (test-time-augmentation). I looked into the [config](https://github.com/hht1996ok/EA-LSS/blob/main/configs/EALSS/ealss_4x8_10e_nusc_aug_large.py) for the [data pipeline](https://github.com/hht1996ok/EA-LSS/blob/main/configs/_base_/datasets/nusc_tf_aug.py). It shows for `test_cfg` that it also uses multi-scale augmentation, as below:

```
    dict(
        type='MultiScaleFlipAug3D',
        img_scale=img_scale,
        pts_scale_ratio=[0.9, 0.95, 1.0, 1.05, 1.1],
        flip=True,
        #rot_range=[-0.19625, 0.0, 0.19625],
        #translation_std=[-0.25, -0.125, 0.0, 0.125, 0.25],
        pcd_horizontal_flip=True,
        pcd_vertical_flip=True,
        transforms=[
            dict(
                type='GlobalRotScaleTransBEV',
                resize_lim=(1.0, 1.0),
                rot_lim=(0.0, 0.0),
                trans_lim=0.0,
                is_train=False
            ),
            dict(type='RandomFlip3DBEV'),
            dict(type='MyResize', img_scale=img_scale, keep_ratio=True),
            dict(type='MyNormalize', **img_norm_cfg),
            dict(type='MyPad', size_divisor=32),
            dict(
                type='DefaultFormatBundle3D',
                class_names=class_names,
                with_label=False),
            dict(type='Collect3D', keys=['points', 'img', 'radar', 'lidar_aug_matrix'])
        ])
```

Could you please confirm that the above augmentations are TTA augmentations, and also if the 71.2 mAP performance on nuScenes val set is obtained with these multi-scale TTA augmentations?

Also, does the test set submission use the same TTA augmentations?

Thanks!

**Comment by hht1996ok** [comment]
Hi @anuj-sharma-19 ,
Thank you for your question, sorry for not commenting the TTA section in the code.
1. 71.2 mAP is the result of EA-LSS in the validation set when TTA and model ensemble are not used, and mAP is increased by 1.0+% after TTA is added.
2.  We used TTA and model ensemble in the  submission version of the test set, and the actual results are the same as the README.

**Comment by anuj-sharma-19** [comment]
Hi @hht1996ok , thanks a lot for the quick reply and confirming the results.

Sorry, I have a couple of more questions:
1. When you apply TTA on val/ test, does it only apply on multi-scale + flip augmentation (as in the above test_cfg) or also rotation etc? Could you please share details on this?
2. If possible, could you also please share details on the ensemble configuration?

Thanks!

**Comment by hht1996ok** [comment]
Hi @anuj-sharma-19 ，
1. We Only used multi-scale and flip augmentation in TTA. Scale parameter is set to [0.9, 0.95, 1.0, 1.05, 1.1] and flipped horizontally and vertically. Instead of NMS, we use Weighted Boxes Fusion.
2. In model ensemble, we use NMS to merge models of different voxel_sizes.

**Comment by anuj-sharma-19** [comment]
That's great, thanks @hht1996ok for all the clarifications :+1:

**Comment by anuj-sharma-19** [comment]
Hi @hht1996ok,

Its me again. I was wondering if there are any plans to also release the checkpoints for the final (fusion) and the intermediate models (lidar, camera) and the config to train the fusion model?

Thanks.
